### PR TITLE
Remove obsolete MEF v1 method of exporting options: [ExportedOption]

### PR DESCRIPTION
**Pull request by DustinCampbell**
This is just the first step to getting Microsoft.CodeAnalysis.Workspaces.Desktop under control. The `[ExportedOption]` attribute is dependent on MEF v1 to allow exporting a field of a class. However, MEF v1 is not portable and there is an official MEF v2 supported way to export options using an `IOptionProvider`.

cc @jasonmalinowski

@Pilchie: Anyone else I should loop in here?

### CODE REVIEW

File: src/EditorFeatures/Core/Options/BraceCompletionOptions.cs [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
     internal static class BraceCompletionOptions
     {
         // This is serialized by the Visual Studio-specific LanguageSettingsPersister
-        [ExportOption]
-        public static readonly PerLanguageOption<bool> EnableBraceCompletion = new PerLanguageOption<bool>(nameof(BraceCompletionOptions), nameof(EnableBraceCompletion), defaultValue: true);
+        public static readonly PerLanguageOption<bool> Enable = new PerLanguageOption<bool>(nameof(BraceCompletionOptions), nameof(Enable), defaultValue: true);
+    }
+
+    [ExportOptionProvider, Shared]
+    internal class BraceCompletionLiteralOptionsProvider : IOptionProvider
```
Comment (Pilchie) [inline src/EditorFeatures/Core/Options/BraceCompletionOptions.cs]: Why `Literal` in the name here?

Comment (DustinCampbell) [inline src/EditorFeatures/Core/Options/BraceCompletionOptions.cs]: Copy-paste fail.

Comment (DustinCampbell) [inline src/EditorFeatures/Core/Options/BraceCompletionOptions.cs]: fixed

**Comment by DustinCampbell** [comment]
retest this please

**Comment by jasonmalinowski** [comment]
Thanks for doing this!

**Pull request merged**